Formula/krew.rb

```diff
@@ -15,15 +17,35 @@ class Krew < Formula
   depends_on "kubernetes-cli"
 
   def install
+    commit = Utils.popen_read("git", "rev-parse", "--short=8", "HEAD").chomp
     ENV["CGO_ENABLED"] = "0"
     # build in local dir to avoid this error:
     # go build: cannot write multiple packages to non-directory /usr/local/Cellar/krew/0.3.2/bin/krew
     mkdir "build"
-    system "go", "build", "-o", "build", "-tags", "netgo", "./cmd/krew/..."
+
+    ldflags = %W[
+      -w
+      -X sigs.k8s.io/krew/pkg/version.gitCommit=#{commit}
+      -X sigs.k8s.io/krew/pkg/version.gitTag=v#{version}
+    ]
+
+    system "go", "build", "-o", "build", "-tags", "netgo",
+      "-ldflags", ldflags.join(" "), "./cmd/krew"
     # install as kubectl-krew for kubectl to find as plugin
     bin.install "build/krew" => "kubectl-krew"
   end
 
+  def caveats
+    <<~EOS
+      To complete the installation, you need to add the following to your
+      ~/.bash_profile, ~/.bashrc or ~/.zshrc.
+
+        export PATH="${PATH}:${HOME}/.krew/bin"
```

Comment:
This is not homebrew specific so it shouldn't be a caveat.

Reply:
Without this, user cannot use the package at all period.

Many other formulas use this field to show less important things, e.g. https://github.com/Homebrew/homebrew-core/blob/master/Formula/fzf.rb#L30-L35

Reply:
That thing you linked is specific to the homebrew install location, that's why it's a caveat. The normal manual won't have homebrew specific instructions. The text you're printing is not specific to homebrew and is therefore not valid as a caveat.

If you find other software that has caveats that aren't specific to homebrew I welcome pull requests to remove it with open arms.

Reply:
OK so I guess caveats are not the right place –but without this, any user journey simply silently won't work.

For example:
- As a user I hear about this tool in a blog post, I find it on Homebrew, I run `brew install krew`. Then I type `krew` &rarr; doesn't work.
- After digging a little and re-reading the blog post, I realize I need to invoke it as `kubectl krew` (this works because the bin is linked as kubectl-krew, which makes this invocation work).
- Then, I install a thing like `kubectl krew install foo`, but now `foo` or `kubectl foo` doesn't work –because I did not complete post-installation instructions (they were not printed by Homebrew).

If we can't add a post-installation message, installation via Homebrew becomes an inferior installation experience than what we have at https://github.com/kubernetes-sigs/krew/#bash-and-zsh. 

In #49428 we've decided that we're ok with having Krew in Homebrew, but we need to make sure the installation experience is equivalent or better. Without a chance to show a post-install message, it's not.

cc: @corneliusweig

Reply:
If the formula can't show any post-install message, this is very sad. Basically any user journey is broken after install as Ahmet just explained.

What other mechanism could you recommend to complete the installation?

Reply:
You could print a warning when people run the tool without having the folder in their path. That benefits everyone. Not just homebrew users.

Reply:
SGTM. We'll handle it in our side.
Removed caveats PTAL.